sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/SqlParser.scala

```diff
@@ -179,10 +180,15 @@ object SqlParser extends AbstractSparkSQLParser with DataTypeParser {
     )
 
   protected lazy val joinedRelation: Parser[LogicalPlan] =
-    relationFactor ~ rep1(joinType.? ~ (JOIN ~> relationFactor) ~ joinConditions.?) ^^ {
+    relationFactor ~
+      rep1(NATURAL.? ~ joinType.? ~ (JOIN ~> relationFactor) ~ joinConditions.?) ^^ {
       case r1 ~ joins =>
-        joins.foldLeft(r1) { case (lhs, jt ~ rhs ~ cond) =>
-          Join(lhs, rhs, joinType = jt.getOrElse(Inner), cond)
+        joins.foldLeft(r1) { case (lhs, n ~ jt ~ rhs ~ cond) =>
+          if (n.isDefined) {
+            NaturalJoin(lhs, rhs, joinType = jt.getOrElse(Inner), cond)
```

Comment:
can't we just create a new join type rather than creating a whole new logical operator?